addons/website/static/src/js/editor/html_editor.js

```diff
@@ -0,0 +1,150 @@
+import { LinkPopover } from "@html_editor/main/link/link_popover";
+import { rpc } from "@web/core/network/rpc";
+import { _t } from "@web/core/l10n/translation";
+import { AutoComplete } from "@web/core/autocomplete/autocomplete";
+import { patch } from "@web/core/utils/patch";
+import { useAutofocus, useChildRef } from "@web/core/utils/hooks";
+import { loadAnchorsLinkPopover } from "../utils";
+
+export class AutoCompleteInLinkpopover extends AutoComplete {
```

Comment:
This whole file hugely lacks comments: you should explain the purpose, you should document the functions, etc. Otherwise I simply cannot make a meaningful review.

Reply:
Yess, my bad :pray: I added some comments at the class and patching. I didn't put more at the functions cause it refactors the `url_autocomplete.js` and `autocomplete_with_pages.js` and I didn't change much the functions to load the option source.  :pray: